chia/full_node/mempool.py

```diff
@@ -44,32 +52,42 @@ def remove_spend(self, item: MempoolItem):
         dic = self.sorted_spends[item.fee_per_cost]
         if len(dic.values()) == 0:
             del self.sorted_spends[item.fee_per_cost]
+        self.total_mempool_cost -= item.cost_result.cost
+        assert self.total_mempool_cost >= 0
 
     def add_to_pool(
         self,
         item: MempoolItem,
         additions: List[Coin],
         removals_dic: Dict[bytes32, Coin],
     ):
-        if self.at_full_capacity():
+        """
+        Adds an item to the mempool by kicking out transactions (if it doesn't fit), in order of increasing fee per cost
+        """
+
+        while self.at_full_capacity(item.cost_result.cost):
             # Val is Dict[hash, MempoolItem]
             fee_per_cost, val = self.sorted_spends.peekitem(index=0)
             to_remove = list(val.values())[0]
-            self.remove_spend(to_remove)
+            self.remove_from_pool(to_remove)
 
         self.spends[item.name] = item
 
         # sorted_spends is Dict[float, Dict[bytes32, MempoolItem]]
-        if item.fee_per_cost in self.sorted_spends:
-            self.sorted_spends[item.fee_per_cost][item.name] = item
-        else:
+        if item.fee_per_cost not in self.sorted_spends:
             self.sorted_spends[item.fee_per_cost] = {}
-            self.sorted_spends[item.fee_per_cost][item.name] = item
+
+        self.sorted_spends[item.fee_per_cost][item.name] = item
 
         for add in additions:
             self.additions[add.name()] = item
         for key in removals_dic.keys():
             self.removals[key] = item
+        self.total_mempool_cost += item.cost_result.cost
+
+    def at_full_capacity(self, cost: int) -> bool:
```

Comment:
Consider `capacity`, which would simply return `return self.total_mempool_cost - self.max_size_in_cost`.

Reply:
:+1:

Reply:
Actually, total mempool cost will always be below the max size